### PR TITLE
Slack BYOI

### PR DESCRIPTION
**Pull request by liorblob**
## Status
In Progress

## Related Issues
https://github.com/demisto/etc/issues/11439
https://github.com/demisto/etc/issues/17532

fixes https://github.com/demisto/etc/issues/17914
fixes https://github.com/demisto/etc/issues/9224
fixes https://github.com/demisto/etc/issues/17332
fixes https://github.com/demisto/etc/issues/15864
fixes https://github.com/demisto/etc/issues/18425

## Required version of Demisto
5.0

## Must have
- [X] Tests
- [ ] Documentation (with link to it)
- [X] Code Review

### CODE REVIEW

**Comment by glicht** [comment]
@liorblob flake8 is failing.

File: Integrations/Slack/Slack.py [hunk 1]
```diff
@@ -0,0 +1,1095 @@
...
+        'slack-set-channel-topic': slack_set_channel_topic,
+        'close-channel': close_channel,
+        'slack-close-channel': close_channel
+    }
+
+    try:
+        command_func = commands[demisto.command()]
+        command_func()
+    except Exception as e:
+        return_error(str(e))
```
Comment (glicht) [inline Integrations/Slack/Slack.py]: We probably want to add a log of the stack trace here too. It is going to be very hard to understand errors without this.

File: Integrations/Slack/Slack_description.md [hunk 2]
```diff
@@ -0,0 +1 @@
+ To generate tokens, browse to [demistobot](https://demistobot.demisto.com) and follow the instructions
```
Comment (glicht) [inline Integrations/Slack/Slack_description.md]: Change to point to new oproxy url: https://oproxy.demisto.ninja/slack

File: Releases/LatestRelease/SlackV2.md [hunk 3]
```diff
@@ -0,0 +1 @@
+New Slack integration
```
Comment (glicht) [inline Releases/LatestRelease/SlackV2.md]: Add also what new features are available

**Comment by liorblob** [comment]
@kirbles19 Please review

**Comment by kirbles19** [comment]
@liorblob Please include a link to the files we're supposed to review.

**Comment by liorblob** [comment]
@kirbles19 `SlackV2.md`, `SlackV2.yml`, `SlackV2.py`.

**Comment by liorblob** [comment]
@kirbles19 @ygerowitz
The following messages need to be reviewed in `Slack.py`:
**Direct messages to the user**:
'You are not allowed to create incidents.' - If it's an external user trying to create incidents while the setting for that is turned off.
496 - 'No other properties other than json should be specified.' - If a user is trying to create incidents while providing JSON, no other properties(name/type) should be specified.
509 - 'Please specify arguments in the following manner: name=<name> type=[type] or json=<json>.' - The user must specify a name while creating incidents, they may get this error if they're sending something like - `create incident 123`.
533 - 'Successfully created incident {}.\n View it on: {}#/WarRoom/{}' - Incident successfully created in Demisto.

**Module health errors**
These errors show up here, if the integration fails to retrieve messages from slack:
image
566 - 'Slack API has thrown an error. Code: {}, Message: {}.'
622 - 'Error occurred while listening to Slack: {}'
419 - 'Slack client raised an exception: {}'

**War Room entries**
Entries that the user sees on command executions (line numbers):
233
302
304
306
350
725
742
759
779
793
795
964
1002
1006
1019
1024
1038

**Comment by liorblob** [comment]
@ygerowitz Also please review the integration description (`Send messages and notifications to your Slack Team.`)

**Comment by kirbles19** [comment]
@liorblob @Itay4 All files edited. Yaniv did the work, I just updated.

**Pull request merged**